metis/lib/server/controllers/metis_controller.rb

```diff
@@ -54,20 +53,24 @@ def file_hashes_with_calculated_paths(bucket:, offset:, limit:, files:)
       }
     end
 
-    def folder_hashes_with_calculated_paths(bucket:, offset:, limit:, all_folders: nil, target_folders: nil)
+    def folder_hashes_with_calculated_paths(bucket:, limit: nil, offset: 0, all_folders: nil, target_folders: nil)
       # Calculate the folder_path, instead of
       #   doing it in the database.
       folder_path_calc = Metis::FolderPathCalculator.new(all_folders: all_folders, bucket: bucket)
 
       # Sorting folders by depth level makes some subsequent calculations simpler,
       #   especially when not paging. Shallow -> deep
-      sorted_folders = all_folders ? sort_folders_by_depth(
-        all_folders: all_folders,
-        limit: limit,
-        offset: offset
-      ) : target_folders
+      if all_folders
+        sorted_folders = sort_folders_by_depth(
```

Comment:
A bit messy, but basically on a paged query first sort by the order_by given, but on a full query also sort by depth.